pkg/artifactory/resource_artifactory_certificate.go

```diff
@@ -0,0 +1,194 @@
+package artifactory
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"strings"
+
+	"github.com/atlassian/go-artifactory/v2/artifactory"
+	v1 "github.com/atlassian/go-artifactory/v2/artifactory/v1"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArtifactoryCertificate() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCertificateCreate,
+		Read:   resourceCertificateRead,
+		Update: resourceCertificateUpdate,
+		Delete: resourceCertificateDelete,
+		Exists: resourceCertificateExists,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"alias": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"content": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"fingerprint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"issued_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"issued_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"issued_to": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"valid_until": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+
+		CustomizeDiff: func(d *schema.ResourceDiff, v interface{}) error {
+			fingerprint, err := calculateFingerPrint(d.Get("content").(string))
```

Comment:
There are a bunch of helpers in the base artifactory go package. You should probably use `getStringRef` instead

Reply:
The code is using a `*schema.ResourceDiff` object, rather than a `*schema.ResourceData` object so there isn't currently a `getStringRef` method defined on that, (could do, however it's only going to be used for these two lines, not sure it's worth it?)

Reply:
Ah cool, my bad